who_l3_smart_tools/core/logical_models/logical_model_generator.py

```diff
@@ -103,20 +104,40 @@
 inflect_engine = inflect.engine()
 
 
+# pylint: disable=too-many-locals,too-few-public-methods
 class LogicalModelAndTerminologyGenerator:
+    """
+    Class for generating FSH logical models and terminologies based on an input Excel file.
+
+    Attributes:
+        input_file (str): The path to the input Excel file.
+        output_dir (str): The directory where the generated FSH files will be saved.
+        models_dir (str): The directory within the output directory where the
+            logical models will be saved.
+        codesystem_dir (str): The directory within the output directory
+            where the code systems will be saved.
+        valuesets_dir (str): The directory within the output directory
+            where the value sets will be saved.
+        invariants_dict (defaultdict): A dictionary to store the
+            invariants for each logical model.
```

Comment:
This isn't quite right. We number invariants based on the process set they fall in (e.g., HIV-A-1, HIV-A-2, HIV-D-1). This stores the counter that tracks the "next" number to use for each prefix. It's not intended to be accessed outside the class, so maybe I should've marked it as such.